src/lib.rs

```diff
@@ -92,9 +92,21 @@ impl Default for Type {
     }
 }
 
-/// Builder for the generator.
-#[derive(Default, Clone, Debug, PartialEq)]
-pub struct Builder<F, V>
+/// This is the error type for the `Builder::build` function.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum ConfigurationError {
+    RadiusInvalid,
+    RadiusUnspecified,
+    RadiusMultiplyDefined,
+    PoissonTypeUnspecified,
+    PoissonTypeMultiplyDefined,
+    SampleCountInvalid,
+    SampleApproximationUnsupported,
+    PoissonTypeUnknownAtRadiusCalculation,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct PoissonConfiguration<F, V>
```

Comment:
Having configuration object way better than what was before. I felt weird having all kinds of methods taking builder as parameter.
I would change the name to just `Configuration` as I have done with `Generator`, `Algorithm` and others.
The idea is that from outside perspective we have `poisson::Configuration` instead `poisson::PoissonConfiguration`.
If the user has conflict with their own type/trait or with another librarys they can always do:
`use poisson::Configuration as PoissonConfiguration;`.